### DISCUSSION TITLE
Required help in FileCacheOption in Ocelot in V23.2.2

### DISCUSSION THREAD

**Discussion by SheruGaur** [ThreeMammals/Ocelot#2054] [General]
Hi @raman-m , 

We are facing some issue regarding cache option, however in  earlier version till v22.0.1  FileCacheOptions was working fine but after the release of v23.2.2 we are facing issue mentioned below in two cases.

**Case 1: V.22.0.1
"FileCacheOptions": { "TtlSeconds": 15, "Region": "europe-central" }** 

It was working fine with the multiple requests. before the latest update 
case2:
After the addition of new parameter named Header in FileCacheOptions there, we are facing issue on multiple requests in case one request is already generated and in the same configured cache expiry time if another request come with different values, then we are getting same response as per the first request.

**Case 2: v23.2.2
"FileCacheOptions": {
  "TtlSeconds": 15,
  "Region": "europe-central",
  "Header": "Authorization"
}**

**Can you please guide what are the difference between above mentioned two releases, earlier cache option is working fine with multiple request input with different arguments. **
Do we need header for caching to work with release 23.2.2? or it can work if we do not use header as it doesn't seem to work without configuring header.**

**Comment by raman-m** [comment]
If you know the issue, why didn't you report the issue?

> **"Header": "Authorization"**

Please stop kidding in my repo please!

**Reply by SheruGaur** [reply]
@raman-m  Hi sorry for misunderstanding if you are thinking am kidding in your repo. Actually, we really facing issue in both releases for FileCacheOption. We are grateful if you guide us.

**Reply by raman-m** [reply]
Seems you don't understand the feature... Docs is here: [Configuration](https://ocelot.readthedocs.io/en/latest/features/caching.html#configuration)

> Do we need header for caching to work with release 23.2.2?

No, a header is not crucial; it merely assists in managing the cache value. It should be clear that without a header, the cached value will expire after `TtlSeconds`. However, if a header is defined, then the cached value will expire immediately with the new header value due to the new caching key.

> or it can work if **we do not use header as it doesn't seem to work without configuring header**.

It should function without the header. If you suspect there is a breaking change introduced with the release of feature in version [23.0.0](https://github.com/ThreeMammals/Ocelot/releases/tag/23.0.0), please could you officially report the bug? The new caching feature by header was added in the commit e92b103644a573a73e3f720b134b9baac86cf145.

**Reply by raman-m** [reply]
The problem is addressed in #2058
